src/mock/quic_p2p/node.rs

```diff
@@ -163,16 +163,16 @@ impl Node {
                         peer: Peer::new(peer_type, src),
                     })
                 }
+                self.send_pending_messages(src);
             }
             Packet::ConnectSuccess => {
                 if self.peers.insert(src, ConnectionType::Normal).is_none() {
-                    let _ = self.bootstrap_cache.insert(NodeInfo::from(src));
-                    self.send_pending_messages(src);
-
                     self.fire_event(Event::ConnectedTo {
                         peer: Peer::node(src),
                     });
                 }
+                let _ = self.bootstrap_cache.insert(NodeInfo::from(src));
```

Comment:
Why is it outside the if?

Reply:
Because it's harmless if we insert it twice, and it would be incorrect to not insert it at all just because we were already connected.

Reply:
We agreed on another patch - I'll resolve the conversation.